analyses/tmb-vaf-longitudinal/01-preprocess-data.Rmd

```diff
@@ -0,0 +1,175 @@
+  left_join(pbta, by = "Kids_First_Participant_ID") %>%
+  filter(!(experimental_strategy == "RNA-Seq")) %>%
+  mutate(tumor_descriptor = case_when(grepl("Primary Tumor", tumor_descriptor) ~ "Diagnosis", 
+                                      grepl("Initial CNS Tumor", tumor_descriptor) ~ "Diagnosis", 
+                                      grepl("Progressive Disease Post-Mortem", tumor_descriptor) ~ "Deceased", 
+                                      TRUE ~ tumor_descriptor)) 
+```
+
+# Create maf file for autopsy samples
+```{r maf_autopsy}
+# We will filter the df to keep paired smaples with Deceased timepoint (autopsy samples).
+# There are patient cases with more than one time points in addition to Deceased, e.g. Diagnosis and Progressive.
+# There are also multiple biospecimen samples per timepoint and patient case.
+# We will create unique identifiers for each case that will enable to create one plot for each case.
+maf_autopsy <- patient_list_metadata %>%
+  mutate(Deceased = case_when(grepl("Deceased", descriptors) ~ "Deceased")) %>%
+  filter(Deceased == "Deceased") %>% 
+  left_join(maf, by = "Kids_First_Biospecimen_ID") %>%
+  filter(!is.na(Kids_First_Participant_ID),
+         !is.na(VAF),
+         !VAF == 0, 
+         !gene_protein == "Unknown") %>%
+  select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, descriptors, tumor_descriptor, experimental_strategy, Hugo_Symbol, gene_protein, VAF) %>%
+  mutate(match_id = paste(Kids_First_Biospecimen_ID, tumor_descriptor, gene_protein, VAF, sep = "_"),
+         match_id_bs = paste(tumor_descriptor, Kids_First_Biospecimen_ID, sep = "_"),
+         timepoints_deceased = match_id_bs,
+         timepoints_deceased = case_when(grepl("Diagnosis", match_id_bs) ~ "1",
+                                         grepl("Recurrence", match_id_bs) ~ "2",
+                                         grepl("Progressive", match_id_bs) ~ "3",
+                                         TRUE ~ timepoints_deceased),
+         timepoints_other = match_id_bs,
+         timepoints_other = case_when(grepl("Deceased", match_id_bs) ~ "4",
+                                      TRUE ~ timepoints_other)) %>%
+  unique() %>%
+  write_tsv(file.path(scratch_dir, "maf_autopsy.tsv"))
+
+# How many patients have paired time samples with autopsy time point present?
+paired_time_samples_autopsy <- length(unique(maf_autopsy$Kids_First_Participant_ID))
+```
+## How many patient cases with autopsy samples in the paired longitudinal data?
+There are `r paired_time_samples_autopsy` autopsy samples out of the total `r paired_time_samples` patient cases.
+
+# Create tmb file for all patient samples with genomic assays
+```{r tmb_all}
+# We will create similar unique identifiers as we did for the maf file.
+# We will also create identifiers for each timepoint (timepoints_deceased and timepoints_other).
+tmb_genomic <- patient_list_metadata %>% 
+  left_join(tmb, by = "Kids_First_Biospecimen_ID") %>%
```

Comment:
```suggestion
  left_join(tmb, by = c("Kids_First_Biospecimen_ID", "experimental_strategy")) %>%
```
since `tmb` and `patient_list_metadata` both have `experimental_strategy` columns, so they are duplicated in `tmb_genomic` (with `.x` and `.y` suffixes). This is one suggestion to prevent that.